plutus-emulator/src/Wallet/Emulator/Types.hs

```diff
@@ -389,12 +388,11 @@ data Event n a where
 type Trace m = Op.Program (Event m)
 
 -- | The state of the emulator itself.
+
 data EmulatorState = EmulatorState {
-    _chainNewestFirst :: Blockchain, -- ^ The current chain, with the newest transactions first in the list.
-    _txPool           :: TxPool, -- ^ The pool of pending transactions.
-    _walletStates     :: Map Wallet WalletState, -- ^ The state of each wallet.
-    _index            :: Index.UtxoIndex, -- ^ The UTxO index, used for validation.
-    _emulatorLog      :: [EmulatorEvent] -- ^ The emulator events, with the newest first.
+    _serverState  :: ServerState,
```

Comment:
I'm a bit surprised to see this in here *without* any kind of monad controlling how we can access it. I guess I'd expect it to be in a separate constraint? Or maybe we run into problems having multiple `MonadState` constraints then...